packages/react-reconciler/src/ReactFiberRoot.new.js

```diff
@@ -104,7 +105,7 @@ export function markRootSuspendedAtTime(
     root.lastSuspendedTime = expirationTime;
   }
 
-  if (expirationTime <= root.lastPingedTime) {
+  if (didCompleteRoot && expirationTime <= root.lastPingedTime) {
     root.lastPingedTime = NoWork;
   }
 
```

Comment:
I wonder if the `lastExpiredTime` assignment just below is similarly suspicious when called outside the commit. Or perhaps not?

Reply:
Looks like wrapping the `lastExpiredTime` in this condition also passes all our tests. Don't know if this is good or bad.